### DISCUSSION TITLE
Hand "Move" function + Smear Tool

### DISCUSSION THREAD

**Discussion by Cryptoconomy** [0Tick/a1111-mini-paint#3] [Ideas]
Just wanted to say this is an awesome extension. My one big frustration is moving the canvas around though. I'm so used to being able to click around with a hand tool using my mouse to drag the canvas to the location I want to work on. I'm honestly not sure how to move the canvas after using this for a while, to the point that I *almost* can't use it. It would be easier to ignore except that then the zoom in/out with scroll (on Macbook) is super aggressive. So getting the right zoom and view position is difficult. When I land on the right location it's great, creating layers is intuitive, and it does like 90% of what I would want in a simple compositing tool on Automatic1111. But navigating the canvas is hard and if there are particular controls, they feel hidden. I resorted to just hitting all the relevant seeming keys and key pairs on my keyboard to see if there was some way I was missing an easy way to just scroll around on the workspace.

Second, another big one for merging the edges of layers was always the smudge tool. Just being able to brush a little bit of the top layer onto the layer below helped to smooth connections between things and I've always found it useful for this sort of stuff.

Will happily donate $50 in sats for the consideration if you have a lightning wallet.

**Comment by 0Tick** [comment]
I will reference you this [issue](https://github.com/viliusle/miniPaint/issues/366) from the original miniPaint. It looks like the author won’t implement something in this direction but I will look into what I can do and notify you.

**Comment by 0Tick** [comment]
https://github.com/0Tick/a1111-mini-paint/assets/96893104/9e2085d7-a434-4005-96a0-79e7e039f6db
This is what i came up with. When choosing a to high size while not scrolled in far enough it behaves weird. Should i commit this?
